go.mod

```diff
@@ -1,69 +1,154 @@
 module github.com/golang-migrate/migrate/v4
 
+go 1.17
```

Comment:
What's the motivation behind this change? Is it the grouping of indirect dependencies? We still need to support Go 1.16.

Reply:
1.17 pulls in a lot fewer dependencies -- basically exactly what is needed to run.

You can import libraries written for 1.17 just fine from 1.16 projects:

main.go has the example from the README loading migrations from github into a posgres db:
```golang
func main() {
        m, err := migrate.New(os.Getenv("MIGRATIONS_URL"), os.Getenv("DATABASE_URL"))
        if err != nil {
                log.Fatal(err)
        }

        fmt.Printf("m=%#v\n", m)
}
```
```
❯ go version
go version go1.16.6 linux/amd64
```
```
❯ go run .
m=&migrate.Migrate{sourceName:"github", sourceDrv:(*github.Github)(0xc000030240), databaseName:"postgres", databaseDrv:(*postgres.P
ostgres)(0xc000244000), Log:migrate.Logger(nil), GracefulStop:(chan bool)(0xc00002a310), isLockedMu:(*sync.Mutex)(0xc00001c790), is
GracefulStop:false, isLocked:false, PrefetchMigrations:0xa, LockTimeout:15000000000}
```

Reply:
The [spec](https://golang.org/ref/mod#go-mod-file-go) allows for this to make backwards incompatible changes although that hasn't been done yet. We should follow the spec instead of the implementation. That being said, we can bump this to go 1.17 once support for 1.16 has been dropped.

Reply:
Done.